test/suites/integration/expiration_test.go

```diff
@@ -61,10 +61,19 @@ var _ = Describe("Expiration", func() {
 		node := env.Monitor.CreatedNodes()[0]
 		env.Monitor.Reset()
 
-		// Eventually expect the node to be gone and a new one to come up
-		env.EventuallyExpectNotFound(node)
+		// Eventually the node will be set as unschedulable, which means its actively being deprovisioned
+		Eventually(func(g Gomega) {
+			n := &v1.Node{}
+			g.Expect(env.Client.Get(env.Context, types.NamespacedName{Name: node.Name}, n)).Should(Succeed())
+			g.Expect(n.Spec.Unschedulable).Should(BeTrue())
+		}).Should(Succeed())
```

Comment:
What if the node is gone and this check never succeeds?

Reply:
Why would that happen? We pre-spin so it should always make it here